src/test/shell/bazel/bazel_symlink_test.sh

```diff
@@ -308,4 +415,135 @@ EOF
   assert_contains "input link is /somewhere/in/my/heart" bazel-bin/a/a.file
 }
 
+function test_symlink_created_from_symlink_action() {
+  mkdir -p a
+  cat > a/a.bzl <<'EOF'
+def _a_impl(ctx):
+    target = ctx.actions.declare_file(ctx.label.name + ".target")
+    ctx.actions.write(
+        output = target,
+        content = "Hello, World!",
+    )
+
+    symlink = ctx.actions.declare_file(ctx.label.name + ".link")
+    ctx.actions.symlink(
+        output = symlink,
+        target_file = target,
+    )
+    return DefaultInfo(files = depset([symlink]))
+
+a = rule(implementation = _a_impl)
+EOF
+
+  cat > a/BUILD <<'EOF'
+load(":a.bzl", "a")
+
+a(name="a")
+EOF
+
+  bazel build //a:a || fail "build failed"
+  assert_contains "Hello, World!" bazel-bin/a/a.link
```

Comment:
Add an assertion that the link file is in fact a symlink (as opposed to a copy).